### PR TITLE
Improve import declaration recovery

### LINKED ISSUE #27026: Invalid diagnostic message for the incomplete import statement
**Description:**
Consider the following incomplete import statement

`import ballerina/module1 a<cursor>`

When this is resolved and at the semantic phase, the compiler gives the following diagnostic.
`cannot resolve module 'ballerina/module1.a as a'`
Also, there are difficulties to suggest the as keyword for the particular context due to the resolving

**Affected Versions:**
slp5 at least

### PR DESCRIPTION
**Pull request by lochana-chathura**
## Purpose
$subject.

Fixes #27026

## Approach
N/A

## Samples
N/A

## Remarks
N/A

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [x] Added necessary documentation  
  - [x] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java [hunk 1]
```diff
@@ -249,11 +249,12 @@
     private static final ParserRuleContext[] IMPORT_VERSION =
             { ParserRuleContext.VERSION_KEYWORD, ParserRuleContext.AS_KEYWORD, ParserRuleContext.SEMICOLON };
 
-    private static final ParserRuleContext[] IMPORT_DECL_RHS = { ParserRuleContext.SLASH, ParserRuleContext.DOT,
-            ParserRuleContext.VERSION_KEYWORD, ParserRuleContext.AS_KEYWORD, ParserRuleContext.SEMICOLON };
+    private static final ParserRuleContext[] IMPORT_DECL_RHS =
```
Comment (rdulmina) [inline compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java:252]: Should we change this to something like `IMPORT_DECL_FIRST_IDENTIFIER_RHS`?

**Comment by mohanvive** [comment]
@lochana-chathura Shall we resolve the conflicts and handle the review comments, please?

**Comment by mohanvive** [comment]
@lochana-chathura Can you please respond to the review comments and handle the conflicts.

**Comment by lochana-chathura** [comment]
> @lochana-chathura Can you please respond to the review comments and handle the conflicts.

On it.

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/AbstractParserErrorHandler.java [hunk 2]
```diff
@@ -299,6 +301,22 @@ protected Result seekInAlternativesPaths(int lookahead, int currentDepth, int cu
...
+            return false;
+        }
+
+        if (result.solution == null) {
+            return true;
+        }
+
+        return result.solution.action != Action.REMOVE;
+    }
+
```
[the remark this answers is not in the document]
Comment (lochana-chathura) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/AbstractParserErrorHandler.java:319]: Can have a remove action with result.matches = LOOKAHEAD_LIMIT - 1 right?
i.e. 4 matches and 1 fix with remove

[the remark this answers is not in the document]
Comment (rdulmina) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/AbstractParserErrorHandler.java:346]: existing check in L#276 does that right?

[the remark this answers is not in the document]
Comment (lochana-chathura) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/AbstractParserErrorHandler.java:346]: The thing is, with that logic we exit when we found  4 matches and 1 fix with REMOVE. But, we shouldn't.

**Comment by lochana-chathura** [comment]
@rdulmina Shall we merge this PR if everything's ok?

**Review by rdulmina — APPROVED** [review]

**Pull request merged**